src/IO/InputStreamFromROOTFile.cxx

```diff
@@ -163,6 +164,51 @@ InputStreamFromROOTFile::set_up(const std::string & header_path)
     return Succeeded::yes;
 }
 
+bool
+InputStreamFromROOTFile::check_brentry_randoms_scatter_energy_conditions(Long64_t brentry)
+{
+  if (brentry < 0)
+    return false;
+
+  // Scatter event condition.
+  if (this->exclude_scattered) {
+    if (br_comptonPhantom1->GetEntry(brentry) == 0)
+      return false;
```

Comment:
Previously these `br_comptonPhantom1->GetEntry(brentry) == 0` checks would `return Succeeded::no;`. With this change, this will now result in a `continue;` in the daughter class, which causes the next entry in the file to be processed. 

I really need the ability to return 3 things, what about?
 - `return 0;` - all good
 - `return 1;` - Event failed the conditions
 - `return 2;` - `GetEntry()== 0`

Reply:
`GetEntry()== 0` is really an `error`, is that correct? The new STIR policy is then to use `error` (which will throw an exception) as opposed to returning `Succeeded::no` (and hoping that the user will check, which half the time they don't).  Possibly that solves your problem. If not, feel free to create an `enum` (I guess `protected` inside here) such that the names tell people what's going on.

Reply:
I think yes, but also no. https://root.cern.ch/doc/master/classTBranch.html#a4f9c8c55ab275930f2fc2f9262e31f91
The key line here is 
> The function returns the number of bytes read from the input buffer. If entry does not exist, the function returns 0. If an I/O error occurs, the function returns -1.

STIR does not check for an I/O error and therefore we really need:
```
if (br_*->GetEntry(brentry) <= 0); 
  error("I have errored!");  // Report an error message.
```